parser/src/powdr.lalrpop

```diff
@@ -450,8 +450,8 @@ BoxedExpression: Box<Expression> = {
 }
 
 LambdaExpression: Box<Expression> = {
-    <kind:FunctionKind> "||" <body:BoxedExpression> => Box::new(Expression::LambdaExpression(LambdaExpression{kind, params: vec![], body})),
-    <kind:FunctionKind> "|" <params:ParameterList> "|" <body:BoxedExpression> => Box::new(Expression::LambdaExpression(LambdaExpression{kind, params, body})),
+    <start:@L> <kind:FunctionKind> "||" <body:BoxedExpression> => Box::new(Expression::LambdaExpression(ctx.source_ref(start), LambdaExpression{kind, params: vec![], body})),
```

Comment:
> Could be `LambdaExpression{kind, praams: vec![], body}.into().with_source_ref(ctx.source_ref(start))`

I tried to implement it that way, but Rust asks me to give a fully qualified path to specify the expected types, and the result is much uglier.

Reply:
What about writing a parser helper function?
```
fn to_expr_with_source_ref<T: Into<Expression>>(inner_expr: T, offset: usize) -> Box<Expression> {
  let mut e = Expression::from(inner_expr);
  e.set_source_ref(offset);
  Box::new(e)
}
```
Also if this is the only use of `set_source_ref`, I would prefer to inline its current definition, because we should not need to set the source ref that often, I think...